src/lib.rs

```diff
@@ -13,24 +13,103 @@
 // under the License.
 
 extern crate fuse;
+extern crate libc;
 #[macro_use] extern crate log;
+extern crate time;
 
+use std::collections::HashMap;
 use std::ffi::OsStr;
 use std::io;
 use std::path::Path;
+use std::sync::{Arc, Mutex};
+use time::Timespec;
+
+mod nodes;
+
+// TODO(jmmv): Make configurable via a flag and store inside SandboxFS.
+pub const TTL: Timespec = Timespec { sec: 60, nsec: 0 };
 
 /// FUSE file system implementation of sandboxfs.
 struct SandboxFS {
+    /// Mapping of inode numbers to in-memory nodes that tracks all files known by sandboxfs.
+    nodes: Arc<Mutex<HashMap<u64, Arc<nodes::Node>>>>,
 }
 
 impl SandboxFS {
     /// Creates a new `SandboxFS` instance.
     fn new() -> SandboxFS {
-        SandboxFS {}
+        let root = {
+            let inode = fuse::FUSE_ROOT_ID;
+            let now = time::get_time();
+            let uid = unsafe { libc::getuid() } as u32;
+            let gid = unsafe { libc::getgid() } as u32;
+            nodes::Dir::new_empty(inode, inode, now, uid, gid)
+        };
+
+        let mut nodes = HashMap::new();
+        nodes.insert(root.inode(), root);
+        SandboxFS {
+            nodes: Arc::from(Mutex::from(nodes)),
```

Comment:
Generally one uses ::new instead of ::from. Is there a reason for the latter?

Reply:
Just me not knowing any better... changed now, but what's the difference? Just idiomatic?

Reply:
`from` implementations sometimes have coercion semantics that you might not want to invoke. `new` is always direct construction without any coercion.